### PR TITLE
Refactored form tester page hooks to support post-hooks

### PR DESCRIPTION
**Pull request by U-DON**
### Background
Previously, page hooks would cause the secondary aXe check on pages to run at the wrong time.

Since page hooks needed to explicitly click the "Continue" button to move to the next page, and that secondary aXe check was set to automatically run after page hooks, the check would incorrectly run on the next page.

### Solution
This refactor fixes that bug, but also reduces the amount of boilerplate needed in page hooks and allows for a way to override the default behavior.
- The aXe checks will still automatically run, but the "Continue" button will also automatically be clicked, so that should no longer be included in page hooks.
- To override that automatic click, there willl now be a context object provided as the first argument to page hooks. It contains an `afterHook` helper function that takes a function and runs that instead of the default behavior of clicking "Continue".
- Docs have been updated (with examples) to explain how to use the override.

## Testing done
Tests are still passing. Observed the Cypress test runner log to verify that aXe checks are happening in the right place.

## Acceptance criteria
- [ ] The form tester should run an aXe check at the beginning and end of processing a form page.
- [ ] The form tester should provide helpful default behavior for processing a form page and also a mechanism to override the defaults.

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by Mottie — APPROVED** [review]
LGTM! 🚀

**Comment by U-DON** [comment]
Could I get reviews from the following? 🙏 
- @department-of-veterans-affairs/vsa-ebenefits-frontend
- @department-of-veterans-affairs/vsa-bam-2-frontend
- @department-of-veterans-affairs/vsa-caregiver-frontend

It should be quick; the changes to your tests are minimal.

**Review by micahchiang — APPROVED** [review]

File: src/platform/testing/e2e/cypress/support/form-tester/index.js [hunk 1]
```diff
@@ -126,40 +126,66 @@ const addNewArrayItem = $form => {
...
-  cy.axeCheck();
-
   cy.location('pathname', NO_LOG_OPTION).then(pathname => {
-    if (pathname.endsWith('review-and-submit')) {
-      performPageActions(pathname, false);
+    performPageActions(pathname);
 
-      // Check the privacy agreement box if it exists.
+    if (!pathname.endsWith('/confirmation')) {
+      cy.location('pathname', NO_LOG_OPTION)
```
Comment (U-DON) [inline src/platform/testing/e2e/cypress/support/form-tester/index.js:169]: It's possible the submission timeout could still happen, but it's rare. Going to keep on eye on tests and re-introduce the extended timeout if it re-appears.

**Review by Clark-W-Griswold — APPROVED** [review]
👍

**Pull request merged**